src/Build/Construction/ProjectItemDefinitionGroupElement.cs

```diff
@@ -30,7 +30,7 @@ internal ProjectItemDefinitionGroupElement(ProjectItemDefinitionGroupElementLink
         /// <summary>
         /// Initialize a parented ProjectItemDefinitionGroupElement
         /// </summary>
-        internal ProjectItemDefinitionGroupElement(XmlElement xmlElement, ProjectRootElement parent, ProjectRootElement containingProject)
+        internal ProjectItemDefinitionGroupElement(XmlElement xmlElement, ProjectElementContainer parent, ProjectRootElement containingProject)
```

Comment:
We should probably add a check to make sure the parent isn't a Target, since we still reject ItemDefinitionGroups under targets.

Reply:
In the function ParseProjectTargetElement，there is https://github.com/dotnet/msbuild/blob/cc3db358d34ad4cd1ec0c67e17582d7ca2a15040/src/Build/Evaluation/ProjectParser.cs#L629-L631, before to call the Itemdefinationgroup constructor, the error "MSB4163: <ItemDefinitionGroup> is not allowed inside a target" has been thrown when parse the child item under the target element.